File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        
        n = len(nums)


        prefix_pro = [1]*n
        suffix_pro = [1]*n

        for i in range(1,n):

            prefix_pro[i] = prefix_pro[i-1] * nums[i-1]

        for i in range(n-2,-1,-1):

            suffix_pro[i] = suffix_pro[i+1] * nums[i+1]

        ans = []

        for i in range(n):

            ans.append(prefix_pro[i] * suffix_pro[i])

        
        return ans

        
        

File: Data Structures & Algorithms/maximum-subarray/submission-0.go


func max(a,b int) int {

	if a > b {

		return a
	}

	return b
}

func maxSubArray(nums []int) int {



	ans := nums[0]
	currSum := nums[0]

	for i := 1; i < len(nums); i++ {

		currSum = max(nums[i], currSum + nums[i])

		ans = max(ans, currSum)

		

	}


	return ans

	
    
}
